apps/gateway/lib/gateway/controllers/arena/currency_controller.ex

```diff
@@ -1,25 +1,29 @@
-defmodule Gateway.Controllers.Users.CurrencyController do
+defmodule Gateway.Curse.Controllers.Users.CurrencyController do
+  @moduledoc """
+    Controller to control currency changes in users
+  """
+
   use Gateway, :controller
 
+  alias GameBackend.Users
   alias GameBackend.Utils
-  alias GameBackend.Users.Currencies.UserCurrency
   alias GameBackend.Users.Currencies
 
-  @moduledoc """
-    Controller to control currency changes in users
-  """
-
   def modify_currency(conn, %{
         "currency_name" => currencty_name,
         "amount" => amount,
-        "user_id" => user_id,
-        "game_name" => game_name
+        "user_id" => user_id
       }) do
-    game_id = Utils.get_game_id(String.to_atom(game_name))
+    {:ok, user} = Users.get_user(user_id)
```

Comment:
This will cause a crash if the user we send doesn't exist. Let's do this inside the `with` block instead!

Reply:
Sure thing